realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java

```diff
@@ -881,6 +881,108 @@ public void contains_caseSensitiveWithNonLatinCharacters() {
     }
 
     @Test
+    public void like_caseSensitive() {
+        final int TEST_OBJECTS_COUNT = 200;
+        populateTestRealm(realm, TEST_OBJECTS_COUNT);
+
+        RealmResults<AllTypes> resultList = realm.where(AllTypes.class)
+                .like("columnString", "*DaTa*").findAll();
+        assertEquals(0, resultList.size());
+
+        resultList = realm.where(AllTypes.class)
+                .like("columnString", "*DaTa*", Case.INSENSITIVE).findAll();
+        assertEquals(TEST_OBJECTS_COUNT, resultList.size());
+
+        resultList = realm.where(AllTypes.class)
+                .like("columnString", "*DaTa 2?").findAll();
+        assertEquals(0, resultList.size());
+
+        resultList = realm.where(AllTypes.class)
+                .like("columnString", "*DaTa 2?", Case.INSENSITIVE).findAll();
+        assertEquals(10, resultList.size());
+
+        resultList = realm.where(AllTypes.class)
+                .like("columnString", "TEST*0").findAll();
+        assertEquals(0, resultList.size());
+
+        resultList = realm.where(AllTypes.class)
+                .like("columnString", "TEST*0", Case.INSENSITIVE).findAll();
+        assertEquals(20, resultList.size());
+    }
+
+    @Test
+    public void like_caseSensitiveWithNonLatinCharacters() {
+        populateTestRealm();
+
+        String flagEmoji = new StringBuilder().append(Character.toChars(0x1F1E9)).toString();
+        String emojis = "ABC" + flagEmoji + "DEF";
+
+        realm.beginTransaction();
+        realm.delete(AllTypes.class);
+        AllTypes at1 = realm.createObject(AllTypes.class);
+        at1.setColumnString("Αλφα");
+        AllTypes at2 = realm.createObject(AllTypes.class);
+        at2.setColumnString("βήτα");
+        AllTypes at3 = realm.createObject(AllTypes.class);
+        at3.setColumnString("δέλτα");
+        AllTypes at4 = realm.createObject(AllTypes.class);
```

Comment:
What is it you would like to test here?  The believe that the only relevant distinction is ASCII/non-ASCII...